cache.js

```diff
@@ -0,0 +1,29 @@
+class Cache {
+  constructor() {
+    this.lastWinners = [];
+    this.lastLosers = [];
+  }
+
+  newWinners(winners) {
```

Comment:
can we name this as an action? setNewWinners

Reply:
i dont think it should be setNewWinners, if we expand the size of the cache, itll be a queue which will be inserting (or some other ds), so I left it as newWinners to indicate the newest winners